neonvm-controller/cmd/main.go

```diff
@@ -219,12 +219,11 @@ func main() {
 	defer ipam.Close()
 
 	vmReconciler := &controllers.VMReconciler{
-		Client:   mgr.GetClient(),
-		Scheme:   mgr.GetScheme(),
-		Recorder: mgr.GetEventRecorderFor("virtualmachine-controller"),
```

Comment:
Maybe it'd be better to just implement a noop `record.EventRecorder` and pass it here?

Reply:
Yes, my thinking is that having a shim would be less moving parts and would be less risky for a hotfix. But if we know for sure that we want to move away from recording k8s events, current diff is better for this.

Reply:
Yeah, I think it's unlikely we're able to support writing to k8s events in the medium term unfortunately :/